File: src/Dignite.FlexFields.Site/Dignite/FlexFields/Site/Matrix/MatrixConfiguration.cs
using System.Collections.Generic;
using Dignite.Abp.FlexFields;

namespace Dignite.FlexFields.Site.Matrix;

public class MatrixConfiguration : FieldConfigurationBase
{
    /// <summary>
    /// The schema: every block type an instance of this field can contain. Reads/writes through
    /// <see cref="FieldConfigurationDictionaryExtensions.GetConfiguration{TConfiguration}"/>'s existing
    /// JSON round-trip fallback (the same mechanism <c>TreeConfiguration.Nodes</c> and
    /// <c>SelectConfiguration.Options</c> already rely on for collection-shaped configuration) - nothing
    /// custom to write here.
    /// </summary>
    public List<MatrixBlockType> BlockTypes {
        get => ConfigurationDictionary.GetConfiguration(MatrixConfigurationNames.BlockTypes, new List<MatrixBlockType>());
        set => ConfigurationDictionary.SetConfiguration(MatrixConfigurationNames.BlockTypes, value);
    }

    public MatrixConfiguration(FieldConfigurationDictionary fieldConfiguration)
        : base(fieldConfiguration)
    {
    }

    public MatrixConfiguration() : base()
    {
    }
}

File: angular/projects/site/src/lib/field-types/matrix/matrix-configuration.ts
import { FormArray } from '@angular/forms';

/**
 * Configuration of a `Matrix` field, shaped for `FormBuilder.group()`. Mirrors `MatrixConfiguration`
 * (`src/Dignite.FlexFields.Site/Dignite/FlexFields/Site/Matrix/MatrixConfiguration.cs`).
 */
export class MatrixConfiguration {
  'Matrix.BlockTypes': unknown = new FormArray<never>([]);
}
